includes/admin/class-wc-rest-payments-files-controller.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * Class WC_REST_Payments_Files_Controller
+ *
+ * @package WooCommerce\Payments\Admin
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * REST controller for files.
+ */
+class WC_REST_Payments_Files_Controller extends WC_Payments_REST_Controller {
+
+	/**
+	 * Endpoint path.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'payments/file';
+
+	/**
+	 * Configure REST API routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			[
+				'methods'             => WP_REST_Server::CREATABLE,
+				'callback'            => [ $this, 'upload_file' ],
+				'permission_callback' => [ $this, 'check_permission' ],
+			]
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<file_id>\w+)',
+			[
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => [ $this, 'get_file' ],
+				'permission_callback' => [],
```

Comment:
Any chance here to keep the permissions checked in generals, but allow free access for branding-related files?
Currently, looks like a potential security issue.

Reply:
Hmm, good point. We might be able to use the `purpose` attribute of the file object returned by Stripe to allow public access to `business_icon` and `business_logo`, and put all other files secured. Wdyt, @dpaun1985?

Reply:
Might be an option! Then we still need to retrieve the file completely, but at least the endpoint itself will be able to decide if to share the content or forbid.

Reply:
As discussed in p1643122246349800-slack-C02C00HCJ8J , we will check the permissions based on `purpose` field